iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -208,27 +208,6 @@ class string
     // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays, hicpp-explicit-conversions)
     string(const char (&other)[N]) noexcept;
 
-    /// @brief conversion constructor for cstring to string which truncates characters if the size is greater than
-    /// the string capacity
-    ///
-    /// @param [in] TruncateToCapacity_t is a compile time variable which is used to distinguish between
-    /// constructors with certain behavior
-    /// @param [in] other is the cstring to convert
-    /// @attention truncates characters if the size is greater than the string capacity
-    ///
-    /// @code
-    ///     #include "iceoryx_hoofs/cxx/string.hpp"
-    ///     using namespace iox::cxx;
-    ///
-    ///     int main()
-    ///     {
-    ///         string<4> fuu(TruncateToCapacity, "abcd");
-    ///     }
-    /// @endcode
-    // TruncateToCapacity_t is a compile time variable to distinguish between constructors
-    // NOLINTNEXTLINE(hicpp-named-parameter, readability-named-parameter)
-    string(TruncateToCapacity_t, const char* const other) noexcept;
```

Comment:
If #1723 is implemented sometime soon, one could also use `string(TruncateToCapacity_t, const iox::string_view* const other)` here. `string_view` has the size of the string stored internally and is the typically C++ idiomatic way to avoid `char*`-only interfaces.